src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -28,11 +29,79 @@
 #include <boost/filesystem/path.hpp>
 #include <iostream>
 #include <sstream>
+#include <bitset>
 
 #define XMAUTILS_MOD "xmautils"
 
 extern XmaSingleton *g_xma_singleton;
 
+namespace xma_core {
+    std::map<XmaSessionType, const char*> sessionMap = {
+        { XmaSessionType::XMA_SCALER, "scaler"},
+        { XmaSessionType::XMA_ENCODER, "encoder"},
+        { XmaSessionType::XMA_DECODER, "decoder"},
+        { XmaSessionType::XMA_FILTER, "filter"},
+        { XmaSessionType::XMA_KERNEL, "kernel"},
+        { XmaSessionType::XMA_ADMIN, "admin"},
+        { XmaSessionType::XMA_INVALID, "invalid"}
+    };
+
+    int32_t finalize_ddr_index(XmaHwKernel* kernel_info, int32_t req_ddr_index, int32_t* ddr_index, char* prefix) {
+        *ddr_index = -1;
```

Comment:
I don't like magic numbers (e.g., numbers without any description).  Yes, I know that is it "obvious" that this is an invalid index, but it would be create to indicate so.  For example:
```
static const int32_t INVALID_INDEX = -1;
*ddr_index = INVALID_INDEX;
```

Reply:
done